### DISCUSSION TITLE
ProxySandbox模式子应用取值源码疑问

### DISCUSSION THREAD

**Discussion by fujingzhi** [umijs/qiankun#1411] [Q&A] [answered]
const value = propertiesWithGetter.has(p)
          ? (rawWindow as any)[p]
          : p in target
          ? (target as any)[p]
          : (rawWindow as any)[p];
ProxySandbox 这种模式子应用通过getter取值时，为什么优先判断propertiesWithGetter.has(p)这个条件，为真为什么就直接从rawWindow取值了？

**Comment by kuitos** [comment, marked as answer]
因为通过 rawWindow.prop 的方式取值才能触发原始的 getter 调用，参考这个 pr 跟 test case https://github.com/umijs/qiankun/pull/641
